### PR TITLE
Allow taking a stream snapshot of a stream that uses TLS

### PR DESCRIPTION
**Pull request by raboof**
Refs #28691

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
Looks reasonable to me 👍

File: akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala [hunk 1]
```diff
@@ -442,7 +442,10 @@ import akka.util.ByteString
     }
   }
 
-  override def receive = inputBunch.subreceive.orElse[Any, Unit](outputBunch.subreceive)
+  override def receive = inputBunch.subreceive.orElse[Any, Unit](outputBunch.subreceive).orElse {
+    case ActorGraphInterpreter.Snapshot =>
+      sender() ! StreamSnapshotImpl(self.path, Seq.empty, Seq.empty)
```
Comment (jrudolph) [inline akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala:447]: The problem here is that the path does not say that this is a TLS stage which will probably lead to some further head scratching. Could we change the naming of TLSModule actors as well?

**Review by jrudolph — APPROVED** [review]
I guess it's better than before, though I suspect that `assertAllStagesStopped` will now produce a basically empty entry for a stream actor with an innocent name which might still be confusing.

**Review by johanandren — APPROVED** [review]

File: akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala [hunk 2]
```diff
@@ -967,7 +967,7 @@ private final case class SavedIslandData(
 
     val props =
       TLSActor.props(maxInputBuffer, tls.createSSLEngine, tls.verifySession, tls.closing).withDispatcher(dispatcher)
-    tlsActor = materializer.actorOf(props, islandName)
+    tlsActor = materializer.actorOf(props, "TLS-for-" + islandName)
```
Comment (jrudolph) [inline akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala:970]: Perfect, thanks :)

**Review by jrudolph — APPROVED** [review]
LGTM

**Comment by jrudolph** [comment]
Great, that's already useful today: https://jenkins.akka.io:8498/job/akka-http-nightly/126/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.13.1,jdk=AdoptOpenJDK%2011/consoleFull

**Pull request merged**